.github/workflows/build.yml

```diff
@@ -21,102 +23,75 @@ defaults:
 
 env:
   CI: true
-  DOCKER_BUILDKIT: 1
   SERVER_IMAGE_NAME: "text-gen-server:0"
 
 jobs:
   build:
     runs-on: ubuntu-latest
+    permissions:
+      packages: write
+      contents: read
     env:
-      BUILDKIT_INLINE_CACHE: 1
+      CACHE_IMAGE: "ghcr.io/ibm/text-gen-server:build-cache"
+      CACHE_REGISTRY: "ghcr.io"
+      SERVER_IMAGE: "ghcr.io/ibm/text-gen-server:latest"  # TODO: don't push final image as a package to ghcr.io
```

Comment:
hmm, why not push the final image?

It'd be great to push both `text-gen-server:latest` and `text-gen-server:${short commit hash}` on main builds

Reply:
I don't think we should use GH packages for images that get pulled externally to honor the quotas
https://docs.github.com/en/billing/managing-billing-for-github-packages/about-billing-for-github-packages